### PR TITLE
[DOCS] Add Empty Types

### LINKED ISSUE #4090: Document `empty` type

### PR DESCRIPTION
**Pull request by goodmind**
Fixes #4090

### CODE REVIEW

File: website/en/docs/types/empty.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+function error(message: string): empty {
+  throw new Error(message)
+}
+```
+
+or never returns:
+
+```js
+// @flow
+function loop(): empty {
```
Comment (goodmind) [inline website/en/docs/types/empty.md]: This actually errors, but I thought that it shouldn't? https://flow.org/try/#0GYVwdgxgLglg9mABAGznADgCgJQC5ECmAtulAJ6IDeAUAJADuAFjMgZlAE4gHZUC+1PkA

Comment (AugustinLF) [reply website/en/docs/types/empty.md]: Hmmm, this function returns `void`, that's perhaps why?

Comment (goodmind) [reply website/en/docs/types/empty.md]: It shouldn't

Comment (goodmind) [reply website/en/docs/types/empty.md]: https://github.com/facebook/flow/issues/4088

Comment (AugustinLF) [reply website/en/docs/types/empty.md]: Oh yes you're right, my bad. But I guess that flow cannot understand that we never exit the loop.

File: website/en/docs/types/empty.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+```js
+//@flow
+type User = {type: 'userSelf'} | {type: 'userForeign', firstName: string}
+
+function handleUser(data: User): string {
+  switch (data.type) {
+    case 'userSelf':
+      return 'Me'
+    case 'userForeign':
+      return user.firstName
```
Comment (Brianzchen) [inline website/en/docs/types/empty.md:33]:
```suggestion
      return data.firstName
```

File: website/en/docs/types/empty.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    default:
+      ;(data.type: empty) // This would error if you forgot to handle case, or misspelled type
```
Comment (Brianzchen) [inline website/en/docs/types/empty.md:35]: If you don't return a string at the end, it doesn't actually work?

https://flow.org/try/#0C4TwDgpgBAqgzhATlAvFA3qSAuKByAVwUQGUIAbAMzwF8oAfDLCXQ4gMQHtEIBLAcwB2eADRRKvRHGAA5AIYBbFlGmJeg-jQBQlAoIDGwXp0FQAFnMEATchHhIAFFbnA5ue4gCUuVev4YtKBUAd15gfTMoJxc5ADpmTwCgoP05BHwiJDIqPGxA5KCeYAJEUzwAWQg8fJS06DYkLh4BYTyCwohi0qhnV1iJKVlFCBqeiEo5AnJgNvaAbmi+5lwIBTBQRIB6TagAFTNeOChgzimrKCREbiheSigQU-Fufk5gKGBOc0sbaFSEMWuCkOcEg5Fs52Y+W0NCAA

**Comment by roryokane** [comment]
This new documentation for the `empty` type is online at https://flow.org/en/docs/types/empty/.

**Bug:** the code blocks are rendered incorrectly:

> Screen Shot 2023-01-08 at 3 08 11 PM

Perhaps the website generator can’t handle the syntax ```` ```js flow-check ```` to start a code block.

**Comment by gkz** [comment]
Fixed with cad9f2531285bf6fcdf6d25e8b5a1e4626b53a14

**Pull request closed without merging**